stdlib/string.ml

```diff
@@ -101,17 +101,13 @@ let trim s =
   else s
 
 let escaped s =
-  let rec needs_escape i =
-    if i >= length s then false else
+  let rec escape s n i =
```

Comment:
Could this function be called `escape_if_needed` for clarity? What it is doing is sort of weird.

(I promise to make this my last nitpicking. On this PR.)

Reply:
Done!